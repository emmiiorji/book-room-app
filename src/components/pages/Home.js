import { useSelector } from 'react-redux/es/exports';
import { FaRegArrowAltCircleDown } from 'react-icons/fa';
import { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import Header from '../Header';
import Stat from '../Stat';
import '../css/home.css';
import SelectYearPopup from '../SelectYearPopup';

export const getObjectLength = (object) => Object.keys(object).length;

const Home = () => {
  const state = useSelector((state) => state);
  const [displayPopup, setDisplayPopup] = useState('none');
  const [countrySelected, setCountrySelected] = useState('');
  const { countries, continents } = state;

  const getTotalHolidays = (countries) => Object.values(countries).reduce((acc, country) => (
    { total: acc.total + country.total_holidays }
  ), { total: 0 }).total.toLocaleString();

  const statsToShow = [
    { name: 'Total Recorded', value: getTotalHolidays(countries) },
    { name: 'Countries', value: getObjectLength(countries) },
    { name: 'Continents', value: getObjectLength(continents) },
  ];

  return (
    <div className="wrapper">
      <div className="overlay">
        <Header />
        <main className="homeMain">
          <h2>Coverage Stats</h2>
          <section className="showStats">
            {statsToShow.map((statToShow) => {
              const { name, value } = statToShow;
              return <Stat key={uuidv4()} name={name} value={value} />;
            })}
          </section>
          <section className="showCountries">
            <div className="filterBar">
              <input type="text" className="searchBox" placeholder="Search Country" />
              <FaRegArrowAltCircleDown className="dropdown" />
            </div>
            <div className="countriesContainer">
              {Object.values(countries).map((country) => {
                const altFlagImage = 'https://res.cloudinary.com/emmii/image/upload/v1656765109/general/inserted-red-color-sticker-label-with-word-unavailable1_vavx1n.jpg';
                return (
                  <button
                    className="showCountry"
                    type="button"
                    key={uuidv4()}
                    onClick={() => { setDisplayPopup('flex'); setCountrySelected(country['iso-3166']); }}
                  >
                    <div>
                      <img className="countryFlag" src={`${country.flag || altFlagImage}`} alt={`${country.country_name}'s flag`} />
                      <p>
                        {country.country_name}
                        :&nbsp;
                        <span style={{ color: 'yellow' }}>{country.total_holidays}</span>
                      </p>
                    </div>
                  </button>
                );
              })}
            </div>
          </section>
        </main>
        <SelectYearPopup
          display={displayPopup}
          hidePopup={() => setDisplayPopup('none')}
          countrySelected={countrySelected}
        />
      </div>
    </div>
  );
};

export default Home;
