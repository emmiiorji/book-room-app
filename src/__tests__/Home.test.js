import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../components/pages/Home';
import store from '../redux/configureStore';

it('component renders properly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
