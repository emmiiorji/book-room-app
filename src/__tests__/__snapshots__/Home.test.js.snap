// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="wrapper"
  data-testid="home"
>
  <div
    className="overlay"
  >
    <header
      className="header"
    >
      <div
        className="hamburger"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": "#fff",
              "height": "30px",
              "width": "30px",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="3"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="21"
            y1="18"
            y2="18"
          />
        </svg>
      </div>
      <div
        className="headerMain"
      >
        <h1
          className="heading"
        >
          Worldwide Holidays
        </h1>
      </div>
    </header>
    <main
      className="homeMain"
    >
      <h2>
        Coverage Stats
      </h2>
      <section
        className="showStats"
      >
        <div
          className="statItem"
        >
          <p
            className="value"
          >
            0
          </p>
          <p
            className="statName"
          >
            Total Recorded
          </p>
        </div>
        <div
          className="statItem"
        >
          <p
            className="value"
          >
            0
          </p>
          <p
            className="statName"
          >
            Countries
          </p>
        </div>
        <div
          className="statItem"
        >
          <p
            className="value"
          >
            0
          </p>
          <p
            className="statName"
          >
            Continents
          </p>
        </div>
      </section>
      <section
        className="showCountries"
      >
        <div
          className="filterBar"
        >
          <input
            className="searchBox"
            onChange={[Function]}
            placeholder="Search Country"
            type="text"
          />
          <select
            className="continentsDropdown"
            name="dropdownFilter"
            onChange={[Function]}
            value="All"
          >
            <option
              value="All"
            >
              All
            </option>
          </select>
        </div>
        <div
          className="countriesContainer"
        />
      </section>
    </main>
    <div
      className="popupOverlay"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="popupContainer"
      >
        <svg
          className="closeBu"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
        <form
          className="selectYearPopup"
          onSubmit={[Function]}
        >
          <p>
            Select Year
          </p>
          <div
            className="rdt"
          >
            <input
              className="form-control"
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              className="rdtPicker"
            >
              <div
                className="rdtYears"
              >
                <table>
                  <thead>
                    <tr>
                      <th
                        className="rdtPrev"
                        onClick={[Function]}
                      >
                        <span>
                          ‹
                        </span>
                      </th>
                      <th
                        className="rdtSwitch"
                        onClick={[Function]}
                      >
                        2020-2029
                      </th>
                      <th
                        className="rdtNext"
                        onClick={[Function]}
                      >
                        <span>
                          ›
                        </span>
                      </th>
                    </tr>
                  </thead>
                </table>
                <table>
                  <tbody>
                    <tr>
                      <td
                        className="rdtYear"
                        data-value={2019}
                        onClick={[Function]}
                      >
                        2019
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2020}
                        onClick={[Function]}
                      >
                        2020
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2021}
                        onClick={[Function]}
                      >
                        2021
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2022}
                        onClick={[Function]}
                      >
                        2022
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtYear"
                        data-value={2023}
                        onClick={[Function]}
                      >
                        2023
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2024}
                        onClick={[Function]}
                      >
                        2024
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2025}
                        onClick={[Function]}
                      >
                        2025
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2026}
                        onClick={[Function]}
                      >
                        2026
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtYear"
                        data-value={2027}
                        onClick={[Function]}
                      >
                        2027
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2028}
                        onClick={[Function]}
                      >
                        2028
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2029}
                        onClick={[Function]}
                      >
                        2029
                      </td>
                      <td
                        className="rdtYear"
                        data-value={2030}
                        onClick={[Function]}
                      >
                        2030
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <input
            type="submit"
            value="See Holidays"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
